pre_processing/seviri_neural_net_preproc.F90

```diff
@@ -0,0 +1,305 @@
+    integer, intent(inout) :: ch1, ch2, ch3, ch4, ch5, ch6, ch7, &
+                              & ch9, ch10, ch11
+
+    ! channel indices
+    ch1 = 1
+    ch2 = 2
+    ch3 = 3
+    ch4 = 4
+    ch5 = 5
+    ch6 = 6
+    ch7 = 7
+    ch9 = 8
+    ch10 = 9
+    ch11 = 10
+end subroutine map_seviri_channels
+
+subroutine get_msg_idx(msg_index, platform, do_nasa)
+    character(len=*), intent(in)    :: platform
+    integer(kind=1),  intent(inout) :: msg_index
+    logical,          intent(in)    :: do_nasa
+
+    select case (trim(platform))
+        case ("MSG1")
+            msg_index = 1
+        case ("MSG2")
+            msg_index = 2
+        case ("MSG3")
+            msg_index = 3
+        case ("MSG4")
+            msg_index = 4
+        case default
+            write(*,*) "Platform ", platform, " not supported with seviri_ml."
+            stop
+    end select
+
+    ! if VIS calibration is not NASA (IMPF) do not apply NASA-to-IMPF
+    ! conversion for ANNs
+    if (.not. do_nasa) then
+        msg_index = 0
+    end if
+end subroutine get_msg_idx
+
+
+subroutine cma_cph_seviri(cview, imager_flags, imager_angles, &
+                          imager_geolocation, imager_measurements, &
+                          imager_pavolonis, skt, channel_info, platform, &
+                          do_nasa)
+
+#ifdef INCLUDE_SEVIRI_NEURALNET
+     use SEVIRI_NEURAL_NET_M
+#endif
+     use preproc_constants_m
+     use imager_structures_m
+     use channel_structures_m
+
+     ! arguments
+     integer,                     intent(in)     :: cview
+     type(imager_flags_t),        intent(in)     :: imager_flags
+     type(imager_angles_t),       intent(in)     :: imager_angles
+     type(imager_geolocation_t),  intent(in)     :: imager_geolocation
+     type(imager_measurements_t), intent(in)     :: imager_measurements
+     type(imager_pavolonis_t),    intent(inout)  :: imager_pavolonis
+     type(channel_info_t),        intent(in)     :: channel_info
+     real(kind=sreal),            intent(in)     :: &
+          skt(imager_geolocation%startx:imager_geolocation%endx, &
+              1:imager_geolocation%ny)
+     character(len=*),             intent(in)     :: platform
+     logical,                     intent(in)     :: do_nasa
+
+     integer(kind=1) :: msg_index
+     logical(kind=1) :: undo_true_reflectances = .false.
+     integer :: ch1, ch2, ch3, ch4, ch5, ch6, ch7, ch9, ch10, ch11
+
+     ! map the channel number to the ORAC array indices
+     call map_seviri_channels(ch1, ch2, ch3, ch4, ch5, ch6, ch7, ch9, ch10, ch11)
+     ! get Meteosat number from platform string
+     call get_msg_idx(msg_index, platform, do_nasa)
+#ifdef INCLUDE_SEVIRI_NEURALNET
+     write(*,*) "PREDICTING COT/CPH"
+     ! run external ANN
+     call seviri_ann_cma(imager_geolocation%nx, &      ! xdim for reshaping
+                imager_geolocation%ny, &               ! ydim for reshaping
+                imager_measurements%data(:,:,ch1), &   ! VIS006
```

Comment:
As @simonrp84 points out, this is going to crash when less than all channels are available (which is the default manner of processing). I imagine the NN requires all of the inputs, so the preprocessor should check that everything is available when the SEVIRI NN is requested.

Reply:
Yes, that's the point I was making- what happens if an extra channel is included?

You are already checking for the case when not enough channels are present.